### PR TITLE
4.0.0 gm

### PR DESCRIPTION
**Pull request by tanner0101**
- [x] Verify README
- [x] Verify supporting GitHub docs
- [x] Verify API docs
- [x] Verify guide docs
- [x] Clean up issues
- [x] Clean up PRs
- [x] Clean up branches
- [x] Remove deprecations

Docs here: https://github.com/vapor/docs/pull/462

Related:
- https://github.com/vapor/jwt/pull/119
- https://github.com/vapor/fluent-kit/pull/241
- https://github.com/vapor/fluent/pull/672

### CODE REVIEW

**Comment by tanner0101** [comment]
Thanks to all the new sponsors! Your name and picture have been added to the README: https://github.com/vapor/vapor/pull/2278/commits/19ea9dc7c8a1fd85829180a29ffb069a08756847

@hjuraev @timothymiko @jstevison @jstorm31 @StefanCosminR @piers12 @iq3addLi @lmcd

Let me know if that looks good. You can choose whichever picture / link you'd like. 

Please contact me on https://discord.gg/vapor to have the `@backer` role added. If you're sponsoring above the $5 tier, let me know so we can arrange the additional rewards.

**Comment by tanner0101** [comment]
I've removed inactive sponsors from the readme: https://github.com/vapor/vapor/pull/2278/commits/a028cc7551b3049e7b7d0bc45b7364c024663a01

@ksmandersen @danielinoa

If this was an accident or I've incorrectly removed you, please let me know. Otherwise, thank you for having been a sponsor.

**Comment by tanner0101** [comment]
Docs here: https://github.com/vapor/docs/pull/462

**Comment by lmcd** [comment]
Thanks for all your hard work. Vapor has been a godsend for a project I'm working on.

Do you think this will happen in a minor point release: https://github.com/vapor/vapor/issues/2235
What's the protocol for breaking API changes?

File: Tests/VaporTests/ServerTests.swift [hunk 1]
```diff
@@ -190,6 +190,9 @@ extension Application {
 final class CustomServer: Server {
     var didStart: Bool
     var didShutdown: Bool
+    var onShutdown: EventLoopFuture<Void> {
+        fatalError()
```
Comment (MrLotU) [inline Tests/VaporTests/ServerTests.swift:194]: @tanner0101 This shouldn't be here?

Comment (MrLotU) [reply Tests/VaporTests/ServerTests.swift:194]: Nevermind. Only just saw it's in a test. 🤦‍♂

**Pull request merged**